homeassistant/components/mqtt/__init__.py

```diff
@@ -404,14 +404,10 @@ async def async_setup_reload_service() -> None:
         async def _reload_config(call: ServiceCall) -> None:
             """Reload the platforms."""
             # Fetch updated manually configured items and validate
-            if (
-                config_yaml := await async_integration_yaml_config(hass, DOMAIN)
-            ) is None:
-                # Raise in case we have an invalid configuration
-                raise HomeAssistantError(
-                    "Error reloading manually configured MQTT items, "
-                    "check your configuration.yaml"
-                )
+            config_yaml = await async_integration_yaml_config(
+                hass, DOMAIN, raise_on_failure=True
+            )
+
```

Comment:
I just think this should raise a ValidationError exception instead (subclass of HomeAssistantError), then the catcher can define if it's a problem or not and if we want to log it.

Ie, standard load of a platform can log it, while a press on validation will not log it.

Reply:
Is `ValidationError` an existing base class? I see it being used as a in import from server libraries. May be we should use another name, or more specific, like `HomeAssistantValidationError`. 
May be I do not understand what you actually mean here.

Reply:
It was just an example name. HomeAssistantValidationError would be fine. But i don't think we need to be that specific in it's. It should be a new exception class like the ServiceError, but geared for this specific use case. Clashing names with some lib should not be that big of a problem.

Reply:
Just a final question on the Exception class. Why can it not be a specialization (subclass) of `HomeAssistantError`. Then we can still catch it, but it it can still be caught as HomeAssistantError. The reason here is to make it easier to replace the HomeAssistantError cases. When used in a service call we can easily catch the exception.

Reply:
It should be a subclass of HomeAssistantError. Always meant that.

Reply:
Re-commited the code, renamed `ServiceError` to `ServiceValidationError`. I added 3 commits.
If you agree I suggest I split up the PR in 3 parts:
1. Add the new Exception, add it to the UI service exception handler (including tests)
2. Allow async_integration_yaml_config to raise (we can change the implementation always to raise).
3. Apply the new solution to the MQTT reload code.